common/cache/lru.go

```diff
@@ -242,16 +252,31 @@ func (c *lru) Size() int {
 	c.mut.Lock()
 	defer c.mut.Unlock()
 
+	if c.activelyEvict {
+		c.evictExpiredItems()
+	}
+
 	return len(c.byKey)
 }
 
+// evictExpiredItems evicts all items in the cache which are expired
+func (c *lru) evictExpiredItems() {
+	for elt := c.byAccess.Back(); len(c.byKey) > 0 && c.isEntryExpired(elt.Value.(*entryImpl), time.Now()); elt = c.byAccess.Back() {
```

Comment:
Maybe it is a bigger change, but I suggest switching from direct time.Now to either stub function now() that returns time.Now that you can mimic, or switching to use of github.com/benbjohnson/clock. That will allow us to not write time related tests and avoid using time.Sleep in tests.